packages/web3-core-method/src/index.js

```diff
@@ -206,8 +207,8 @@ Method.prototype._confirmTransaction = function (defer, result, payload) {
         intervalId = null,
         lastBlock = null,
         receiptJSON = '',
-        gasProvided = (_.isObject(payload.params[0]) && payload.params[0].gas) ? payload.params[0].gas : null,
-        isContractDeployment = _.isObject(payload.params[0]) &&
+        gasProvided = ((typeof payload.params[0] === 'object' && !!payload.params[0]) && payload.params[0].gas) ? payload.params[0].gas : null,
```

Comment:
```suggestion
        gasProvided = ((typeof payload.params[0] === 'object' && payload.params[0]) && payload.params[0].gas) ? payload.params[0].gas : null,
```

Double bang operator not necessary, @spacesailor24 cab we run into weird falsey values?

Reply:
I don't know if it's specifically needed here, but usually the `!!` is used to cast the value to a boolean, but it does seem redundant

Reply:
wait you might be right, i think we should keep this